src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1301,6 +1301,36 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        // TODO-XARCH-AVX512 see how to merge this into below
```

Comment:
You should just be able to merge it into the below. The `if` is just special logic to handle the scenario where you have `AVX` but not `AVX2`.

For `Vector512` we require AVX512F+BW+CD+DQ+VL, so we shouldn't get here if the intrinsic wouldn't be supported. The `(simdSize != 32)` check will ensure it actually gets picked up as intrinsic.